### ISSUE TITLE
Interchromosomal misjoin

### ISSUE THREAD

**Issue by Han-Cao** [chhylp123/hifiasm#651]
Hi, 

I am using hifiasm 0.19.8-r603 to do human genome assembly using hifi-only data. After the assembly, I found there is a interchromosomam misjoin reported by `paftools.js misjoin -c centromeres.bed -e align_to_chm13.paf`.

To troubleshoot the assembly, I aligned the hifi reads to the assembly and check the misjoin position in IGV. In the IGV view, the left part of assembly was aligned to chr10, and the right part of the assembly was aligned to chrX. 

Moreover, the region within the blue window has a lot of supplementary alignments with mismatches and double depth, does it mean that region is a fusion from 2 chromosomes? If I want to manually fix this issue, can I split the assembly into 2 regions: one end with the left blue line, another one start from the right blue line, and drop the region within the blue lines?

Thank you!

![image](https://github.com/chhylp123/hifiasm/assets/34618938/ae91ae2b-0667-473c-844d-8164cdf98577)

**Comment by Han-Cao** [comment]
Hi,

I did more research on this case, and I realized the interchromosomal misjoin could occurs within the black window. I randomly selected one position within the black window and blat the flanking 10kb sequencing to CHM13. The blat result shows the left 5kb sequence is mapped to chr10, while the right 5kb sequencing is mapped to chrX. 

Do you think this is correct way to fix the misjoin issue? Besides, would it possible to avoid such error during assembly?

Thank you!

![image](https://github.com/chhylp123/hifiasm/assets/34618938/b3f8a3d4-82fa-4bd8-97cb-feb331529152)

**Comment by kevfengler227** [comment]
Perhaps the read that was used to build this contig is not begin shown. There should be a read which perfectly matches the assembly. Are you filtering secondary/supplemental alignments in IGV. Are you showing alignments with mapping quality 0?

Another way to troubleshoot is to look into the asm.bp.p_ctg.noseq.gfa file and find the read that used in the tiling path at this location and see why it is not mapping here. I am guessing there bogus or repetitive read that use to create this contig that is not being shown in the visualization.

**Comment by Han-Cao** [comment]
@TDMillar-Biology

Yes, I followed the instructions from HPRC paper, they manually split one assembly in their study, you can find more details in their paper, [hprc-v1.1 tutorial](https://github.com/ComparativeGenomicsToolkit/cactus/blob/master/doc/mc-pangenomes/hprc-v1.1-mc.md#prepare-the-input-assembly-list), and [this issue](https://github.com/ComparativeGenomicsToolkit/cactus/issues/1389).

In my own work, I ran `minigraph` and `paftools.js misjoin` to identify all interchromosomal misjoins
```
minigraph -t 40 -cxasm $ref_chm13 $input > output.paf
paftools.js misjoin -e -c $bed_censat output.paf >  misjoin.txt

# I added -e -c $bed_censat to clean the output, which is different from the code in HPRC paper
```
Then I aligned reads to the assembly and manually check the alignments using IGV. Acrocentric chromosomes are more likely to have real intrachromosomal events (as shown in the HPRC paper), so you need to confirm whether there is an assembly error identified by `paftools.js misjoin`.

Meanwhile, manually checking the alignments also help to identify where the assembly error is. In the above example, the error is within the black window (in the second screenshot), the features of this site are:
- read correctly aligned to its left has many mismatches on its right
- read correctly aligned to its right has many mismatches on its left

Then I just randomly select one position within the black window to split the assembly and ran BLAT on a 5kb window to validate

```
# pos = the position to split
split1_start=$((pos-5000))
split1_end=$pos
split2_start=$((pos+1))
split2_end=$((pos+5001))

samtools faidx ${asm} ${contig}:${split1_start}-${split1_end} > ${asm}.split1.fa
samtools faidx ${asm} ${contig}:${split2_start}-${split2_end} > ${asm}.split2.fa
blat -minScore=4500 -minIdentity=95 $ref_blat ${asm}.split1.fa ${asm}.split1.psl
blat -minScore=4500 -minIdentity=95 $ref_blat ${asm}.split2.fa ${asm}.split2.psl
```

Once I see the 2 splits nearly perfectly aligned to 2 different chromosomes, I consider it is good. Then I corrected the assembly using `samtools faidx`:

```
# contig = the one with assembly error
# pos = the position to split

split1_start=1
split1_end=$pos
split2_start=$((pos+1))
split2_end=$contig_length

keep_contigs=$(awk '{print $1}' ${asm}.fai | grep -v $contig)
samtools faidx ${asm} ${keep_contigs} > clean.fa
echo ">${contig}_sub_0_${pos}" >> clean.fa
samtools faidx ${asm} ${contig}:${split1_start}-${split1_end} | tail -n +2 >> clean.fa
echo ">${contig}_sub_${pos}_${length}" >> clean.fa
samtools faidx ${asm} ${contig}:${split2_start}-${split2_end} | tail -n +2 >> clean.fa
```

Besides, there are different patterns in IGV around assembly error, an easy case is as below. But I remember there were more complex sites, so using blat (or other tools) to align your sequencing to reference genome is necessary to validate.
![image](https://github.com/user-attachments/assets/4cbe2de2-da25-464a-be71-fa1e2f4c5852)
